### DISCUSSION TITLE
Skywalking Installation

### DISCUSSION THREAD

**Discussion by mr-vardhan24** [apache/skywalking#11984] [Q&A]
hello!
I want to deploy skywalking with the backend of the timescale. in Kubernetes v.25
I have tried the steps that followed in the below link but I am not able to deploy, Please share if someone has the proper approach to deploy this.
https://github.com/apache/skywalking-helm

**Comment by wu-sheng** [comment]
If you didn't debug why, I don't think there is something we can do. The oap is verified on Linux and KinD(k8s) for every commit and release.

**Reply by mr-vardhan24** [reply]
Thanks for the reply,
I want to deploy it in my EKS Cluster in which I am already using the timescale, so I want to use timescale as the backend database for the skywalking. 
For the Skywalking installation, I am searching the helm chart which I can apply in my cluster with the values.yaml and make some changes according to my cluster. 
I have followed the documentation but I got direct these type of instructions like this 


`helm install "${SKYWALKING_RELEASE_NAME}" \
  oci://registry-1.docker.io/apache/skywalking-helm \
  --version "${SKYWALKING_RELEASE_VERSION}" \
  -n "${SKYWALKING_RELEASE_NAMESPACE}" \
  --set oap.image.tag=9.2.0 \
  --set oap.storageType=elasticsearch \
  --set ui.image.tag=9.2.0`
  
  
  I can not specify all the info by the --set command so I need a proper helm chart that I can clone and apply the values.yaml as the helm installation procedure. 
  
  please help me with this.

**Reply by wu-sheng** [reply]
We don't have experience of timescale. Sorry. 
But your command is targeting oap.storageType=elasticsearch,  timescale at least is a PostgreSQL mock. 
Ref https://skywalking.apache.org/docs/main/latest/en/setup/backend/storages/postgresql/, maybe `oap.storageType=oap.storageType=`(Just guess)

**Reply by mr-vardhan24** [reply]
Thanks! @wu-sheng 
timescale also uses Postgres
please tell me irrespective of Postgres

so my question is only about how can I get the proper helm chart that I can clone and apply the values.yaml as the helm installation procedure. because in cluster i can not setup skywalking via using --set commands.

**Reply by wu-sheng** [reply]
I am not writing helm. All OAP behaviors are controlled by system environment variables.

I don't know what set means.

**Reply by mr-vardhan24** [reply]
hi @wu-sheng 
hope you are doing well.
i have deployed the skywalking using the docker-compose but I am not able to see the data over skywalking dashboard.

here is my docker-compose.yaml 

`version: '3.3'
services:
  elasticsearch:
    image: docker.elastic.co/elasticsearch/elasticsearch:7.5.0
    container_name: elasticsearch
    restart: always
    ports:
      - 9200:9200
    networks:
      - demo
    environment:
      - discovery.type=single-node
      - bootstrap.memory_lock=true
      - "ES_JAVA_OPTS=-Xms256m -Xmx256m"
    ulimits:
      memlock:
        soft: -1
        hard: -1

        
  oap:
    image: apache/skywalking-oap-server:7.0.0-es7
    container_name: oap
    depends_on:
      - elasticsearch
    links:
      - elasticsearch
    restart: always
    ports:
      - 11800:11800
      - 12800:12800
    networks:
      - demo
    environment:
      SW_STORAGE: elasticsearch7
      SW_STORAGE_ES_CLUSTER_NODES: elasticsearch:9200
  
  ui:
    image: apache/skywalking-ui:7.0.0
    container_name: ui
    depends_on:
      - oap
    links:
      - oap

    networks:
      - demo
    restart: always
    ports:
      - 8080:8080
    environment:
      SW_OAP_ADDRESS: oap:12800


  otel-collector:
    # Specify the image to start the container from
    image: otel/opentelemetry-collector:0.19.0
    # Set the  otel-collector configfile
    command: ["--config=/etc/otel-collector-config.yaml"]
    # Mapping the configfile to host directory
    volumes:
      - ./otel-collector-config.yaml:/etc/otel-collector-config.yaml
    ports:
      - "13133:13133" # health_check extension
      - "12800"       # OpenCensus receiver
    networks:
      - demo

networks:
  demo:
`

and here is the otel-collector-config.yaml

`extensions:
  health_check:
# A receiver is how data gets into the OpenTelemetry Collector
receivers:
  # Set Prometheus Receiver to collects metrics from targets
  # It’s supports the full set of Prometheus configuration
  prometheus:
    config:
      scrape_configs:
        - job_name: 'otel-collector'
          scrape_interval: 10s
          static_configs:
              # Replace the IP to your VMs‘s IP which has installed Node Exporter
            - targets: ['192.168.1.41:9100']
              # - targets: ['vm2:9100']
              #- targets: [‘vm3:9100']

processors:
  batch:

# An exporter is how data gets sent to different systems/back-ends
exporters:
  # Exports metrics via gRPC using OpenCensus format
  opencensus:
    endpoint: "host.docker.internal:12800" # The OAP Server address
    retry_on_failure:
      enabled: true
    insecure: true
  logging:
    logLevel: debug

service:
  pipelines:
    metrics:
      receivers: [prometheus]
      processors: [batch]
      exporters: [logging, opencensus]

  extensions: [health_check]
`

and here the image of running every container 
![image](https://github.com/apache/skywalking/assets/66646221/7abd6e45-9631-4634-bf61-1ba6f59b0a4f)


and here is the logs of the opentelemetry container 


[otl-logs.txt](https://github.com/apache/skywalking/files/14865644/otl-logs.txt)


here is the oap server logs 
[otl-logs.txt](https://github.com/apache/skywalking/files/14865968/otl-logs.txt)
